From 2561b46ffd9c18cc8972b48a111e5dfaa79dfca3 Mon Sep 17 00:00:00 2001
From: Clayton Smith <argilo@gmail.com>
Date: Fri, 8 Dec 2023 19:27:45 -0500
Subject: [PATCH 5/6] Fix flaky mod_range test

Signed-off-by: Clayton Smith <argilo@gmail.com>
---
 kernels/volk/volk_32f_s32f_mod_rangepuppet_32f.h | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/kernels/volk/volk_32f_s32f_mod_rangepuppet_32f.h b/kernels/volk/volk_32f_s32f_mod_rangepuppet_32f.h
index 3e0bc43..3a17859 100644
--- a/kernels/volk/volk_32f_s32f_mod_rangepuppet_32f.h
+++ b/kernels/volk/volk_32f_s32f_mod_rangepuppet_32f.h
@@ -19,7 +19,7 @@ static inline void volk_32f_s32f_mod_rangepuppet_32f_generic(float* output,
                                                              unsigned int num_points)
 {
     volk_32f_s32f_s32f_mod_range_32f_generic(
-        output, input, bound - 3.141f, bound, num_points);
+        output, input, bound - 3.131f, bound, num_points);
 }
 #endif
 
@@ -31,7 +31,7 @@ static inline void volk_32f_s32f_mod_rangepuppet_32f_u_sse(float* output,
                                                            unsigned int num_points)
 {
     volk_32f_s32f_s32f_mod_range_32f_u_sse(
-        output, input, bound - 3.141f, bound, num_points);
+        output, input, bound - 3.131f, bound, num_points);
 }
 #endif
 #ifdef LV_HAVE_SSE
@@ -41,7 +41,7 @@ static inline void volk_32f_s32f_mod_rangepuppet_32f_a_sse(float* output,
                                                            unsigned int num_points)
 {
     volk_32f_s32f_s32f_mod_range_32f_a_sse(
-        output, input, bound - 3.141f, bound, num_points);
+        output, input, bound - 3.131f, bound, num_points);
 }
 #endif
 
@@ -52,7 +52,7 @@ static inline void volk_32f_s32f_mod_rangepuppet_32f_u_sse2(float* output,
                                                             unsigned int num_points)
 {
     volk_32f_s32f_s32f_mod_range_32f_u_sse2(
-        output, input, bound - 3.141f, bound, num_points);
+        output, input, bound - 3.131f, bound, num_points);
 }
 #endif
 #ifdef LV_HAVE_SSE2
@@ -62,7 +62,7 @@ static inline void volk_32f_s32f_mod_rangepuppet_32f_a_sse2(float* output,
                                                             unsigned int num_points)
 {
     volk_32f_s32f_s32f_mod_range_32f_a_sse2(
-        output, input, bound - 3.141f, bound, num_points);
+        output, input, bound - 3.131f, bound, num_points);
 }
 #endif
 
@@ -73,7 +73,7 @@ static inline void volk_32f_s32f_mod_rangepuppet_32f_u_avx(float* output,
                                                            unsigned int num_points)
 {
     volk_32f_s32f_s32f_mod_range_32f_u_avx(
-        output, input, bound - 3.141f, bound, num_points);
+        output, input, bound - 3.131f, bound, num_points);
 }
 #endif
 #ifdef LV_HAVE_AVX
@@ -83,7 +83,7 @@ static inline void volk_32f_s32f_mod_rangepuppet_32f_a_avx(float* output,
                                                            unsigned int num_points)
 {
     volk_32f_s32f_s32f_mod_range_32f_a_avx(
-        output, input, bound - 3.141f, bound, num_points);
+        output, input, bound - 3.131f, bound, num_points);
 }
 #endif
 #endif
-- 
2.39.2

