From 9b2e54b78e65bf7f34e4f8a66c0ed1a2377bec9f Mon Sep 17 00:00:00 2001
From: jdemel <demel@uni-bremen.de>
Date: Thu, 25 Jun 2020 18:35:40 +0200
Subject: [PATCH 9/9] docs: Fix conversion inaccuracy

---
 kernels/volk/volk_32f_s32f_convert_8i.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernels/volk/volk_32f_s32f_convert_8i.h b/kernels/volk/volk_32f_s32f_convert_8i.h
index d08ecd5..4016a52 100644
--- a/kernels/volk/volk_32f_s32f_convert_8i.h
+++ b/kernels/volk/volk_32f_s32f_convert_8i.h
@@ -25,7 +25,7 @@
  *
  * \b Overview
  *
- * Converts a floating point number to a 8-bit char after applying a
+ * Converts a floating point number to a 8-bit int after applying a
  * scaling factor.
  *
  * <b>Dispatcher Prototype</b>
-- 
2.20.1

