From b5992f21b74b129094afe0f2d3e31d5aeb50844c Mon Sep 17 00:00:00 2001
From: Clayton Smith <argilo@gmail.com>
Date: Fri, 8 Dec 2023 21:34:26 -0500
Subject: [PATCH 6/6] Allow for rounding error in float-to-int interleave

Signed-off-by: Clayton Smith <argilo@gmail.com>
---
 lib/kernel_tests.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/lib/kernel_tests.h
+++ b/lib/kernel_tests.h
@@ -104,7 +104,7 @@
     QA(VOLK_INIT_TEST(volk_32fc_deinterleave_32f_x2, test_params))
     QA(VOLK_INIT_TEST(volk_32fc_accumulator_s32fc, test_params.make_absolute(3e-2)))
     QA(VOLK_INIT_TEST(volk_32fc_deinterleave_64f_x2, test_params))
-    QA(VOLK_INIT_TEST(volk_32fc_s32f_deinterleave_real_16i, test_params))
+    QA(VOLK_INIT_TEST(volk_32fc_s32f_deinterleave_real_16i, test_params.make_tol(1)))
     QA(VOLK_INIT_TEST(volk_32fc_deinterleave_imag_32f, test_params))
     QA(VOLK_INIT_TEST(volk_32fc_deinterleave_real_32f, test_params))
     QA(VOLK_INIT_TEST(volk_32fc_deinterleave_real_64f, test_params))
@@ -132,7 +132,7 @@
     QA(VOLK_INIT_TEST(volk_32fc_x2_s32f_square_dist_scalar_mult_32f, test_params))
     QA(VOLK_INIT_TEST(volk_32f_x2_divide_32f, test_params))
     QA(VOLK_INIT_TEST(volk_32f_x2_dot_prod_32f, test_params.make_absolute(1.5e-2)))
-    QA(VOLK_INIT_TEST(volk_32f_x2_s32f_interleave_16ic, test_params))
+    QA(VOLK_INIT_TEST(volk_32f_x2_s32f_interleave_16ic, test_params.make_tol(1)))
     QA(VOLK_INIT_TEST(volk_32f_x2_interleave_32fc, test_params))
     QA(VOLK_INIT_TEST(volk_32f_x2_max_32f, test_params))
     QA(VOLK_INIT_TEST(volk_32f_x2_min_32f, test_params))
